pyproject.toml

```diff
@@ -1,3 +1,77 @@
+[build-system]
+requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
+build-backend = "poetry_dynamic_versioning.backend"
+
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+style = "semver"
+pattern = "default-unprefixed"
+
+[tool.poetry.dependencies]
+python = "^3.7"
+
+# Python lack of functionalities from future versions
+importlib-metadata = { version = "*", python = "<3.8" }
+
+# dev
+pre-commit = { version = "*", optional = true }
+pytest = { version = ">=5.0", optional = true }
+pytest-cov = { version = "2.10.1", optional = true }
+pydot = { version = "*", optional = true }
+neptune = { version = ">=1.0.0", optional = true }
+tensorflow = { version = "*", optional = true }
```

Comment:
Let's move it to non-dev section right below `importlib-metadata`

Reply:
+ Set up proper version specs for `tensorflow` (minimal compatible and up to non-major)